### PR TITLE
Rebase for refactor runtest py

### PR DESCRIPTION
**Pull request by ShakoHo**
@askeing @ypwalter r?

### CODE REVIEW

File: lib/common/environment.py [hunk 1]
```diff
@@ -88,7 +96,8 @@ def __init__(self, test_method_name, test_method_doc, sikuli_script_name=None):
 
     def init_output_dir(self):
         # Init output folder
-        for chk_dir in [self.DEFAULT_OUTPUT_DIR, self.DEFAULT_VIDEO_OUTPUT_DIR, self.DEFAULT_PROFILE_OUTPUT_DIR, self.DEFAULT_IMAGE_DIR,
+        for chk_dir in [self.DEFAULT_OUTPUT_DIR, self.DEFAULT_VIDEO_OUTPUT_DIR, self.DEFAULT_PROFILE_OUTPUT_DIR,
+                        self.DEFAULT_IMAGE_DIR,
                         self.DEFAULT_IMAGE_OUTPUT_DIR, self.DEFAULT_IMAGE_SAMPLE_DIR]:
             if os.path.exists(chk_dir) is False:
```
Comment (askeing) [inline lib/common/environment.py]: `if not os.path.exists(chk_dir):`

File: lib/common/environment.py [hunk 2]
```diff
@@ -88,7 +96,8 @@ def __init__(self, test_method_name, test_method_doc, sikuli_script_name=None):
 
     def init_output_dir(self):
         # Init output folder
-        for chk_dir in [self.DEFAULT_OUTPUT_DIR, self.DEFAULT_VIDEO_OUTPUT_DIR, self.DEFAULT_PROFILE_OUTPUT_DIR, self.DEFAULT_IMAGE_DIR,
+        for chk_dir in [self.DEFAULT_OUTPUT_DIR, self.DEFAULT_VIDEO_OUTPUT_DIR, self.DEFAULT_PROFILE_OUTPUT_DIR,
+                        self.DEFAULT_IMAGE_DIR,
                         self.DEFAULT_IMAGE_OUTPUT_DIR, self.DEFAULT_IMAGE_SAMPLE_DIR]:
```
Comment (askeing) [inline lib/common/environment.py]: Line 100 and 101 can merge into one line.

File: lib/helper/resultHelper.py [hunk 3]
```diff
@@ -70,14 +70,17 @@ def output_result(test_method_name,current_run_result, output_fp, time_list_coun
...
 
-    with open(time_list_counter_fp, "w") as wfh:
-        wfh.write(str(len(result[test_method_name]['time_list'])))
-
+    # output sikuli status to static file
+    with open (time_list_counter_fp) as fh:
+        stat_data = json.load(fh)
+    with open(time_list_counter_fp, "w") as fh:
+        stat_data['time_list_counter'] = str(len(result[test_method_name]['time_list']))
+        json.dump(stat_data, fh)
```
Comment (askeing) [inline lib/helper/resultHelper.py]: If you want to read and then write (overwrite existing contents), you can use `with open(time_list_counter_fp, 'r+') as fh:`.

File: lib/common/environment.py [hunk 4]
```diff
@@ -19,6 +19,7 @@ class Environment(object):
     DEFAULT_CHROME_DRIVER_PATH = os.path.join(DEFAULT_THIRDPARTY_DIR, "chromedriver")
     DEFAULT_SIKULI_PATH = os.path.join(DEFAULT_THIRDPARTY_DIR, "sikulix") if os.path.isfile(os.path.join(DEFAULT_THIRDPARTY_DIR, "sikulix", "runsikulix")) else os.path.join(DEFAULT_THIRDPARTY_DIR)
     DEFAULT_TEST_RESULT = os.path.join(os.getcwd(), "result.json")
+    DEFAULT_STAT_RESULT = os.path.join(os.getcwd(), "stat.json")
     DEFAULT_SIKULI_STATUS_RESULT = os.path.join(os.getcwd(), "sikuli_stat.txt")
```
Comment (askeing) [inline lib/common/environment.py]: If line 22 is replaced line 23, it seems like we can remove this variable.

Comment (ShakoHo) [inline lib/common/environment.py]: yes, line 23 and 24 should remove if we have line 22

File: runtest.py [hunk 5]
```diff
@@ -1,165 +1,154 @@
...
+  runtest.py pilottest <suite.txt> [--max-run=<int>] [--max-retry=<int>] [--keep-browser=<int>] [--profiler=<str>]
+  runtest.py (-h | --help)
+
+Options:
+  -h --help                 Show this screen.
+  --max-run=<int>           Test run max no [default: 40].
+  --max-retry=<int>         Test failed retry max no [default: 15].
+  --keep-browser=<int>      Keep the browser open after test script executed [default: 0]
+  --profiler=<str>          Enabled profiler, current support profiler:avconv,geckoprofiler,harexport,chrometracing,fxall,justprofiler [default: avconv]
+  --online                  Result will be transfer to server, calculated by server
```
Comment (askeing) [inline runtest.py]: Should we setup the online's default value here?

Comment (ShakoHo) [inline runtest.py]: default value is False if we didn't provide ..

Comment (askeing) [inline runtest.py]: Cool! 👍

**Comment by ypwalter** [comment]
r+

**Pull request merged**